sycl/unittests/program_manager/CompileTarget.cpp

```diff
@@ -0,0 +1,249 @@
+  std::copy(CompileTarget.begin(), CompileTarget.end(), Data.data() + 8);
+  UrProperty CompileTargetProperty("compile_target", Data,
+                                   SYCL_PROPERTY_TYPE_BYTE_ARRAY);
+  UrPropertySet PropSet;
+  PropSet.insert(__SYCL_PROPERTY_SET_SYCL_DEVICE_REQUIREMENTS,
+                 {CompileTargetProperty});
+
+  std::vector<unsigned char> Bin(CompileTarget.begin(), CompileTarget.end());
+  // Null terminate the data so it can be interpreted as c string.
+  Bin.push_back(0);
+
+  UrArray<UrOffloadEntry> Entries = makeEmptyKernels({KernelName});
+
+  UrImage Img{SYCL_DEVICE_BINARY_TYPE_NATIVE,          // Format
+              __SYCL_DEVICE_BINARY_TARGET_SPIRV64_GEN, // DeviceTargetSpec
+              "",                                      // Compile options
+              "",                                      // Link options
+              std::move(Bin),
+              std::move(Entries),
+              std::move(PropSet)};
+
+  return Img;
+}
+} // namespace sycl::unittest
+
+class SingleTaskKernel;
+class NDRangeKernel;
+class RangeKernel;
+class NoDeviceKernel;
+
+MOCK_INTEGRATION_HEADER(SingleTaskKernel)
+MOCK_INTEGRATION_HEADER(NDRangeKernel)
+MOCK_INTEGRATION_HEADER(RangeKernel)
+MOCK_INTEGRATION_HEADER(NoDeviceKernel)
+
+static sycl::unittest::UrImage Img[] = {
+    sycl::unittest::generateDefaultImage({"SingleTaskKernel"}),
+    sycl::unittest::generateImageWithCompileTarget("SingleTaskKernel",
+                                                   "spir64_x86_64"),
+    sycl::unittest::generateImageWithCompileTarget("SingleTaskKernel",
+                                                   "intel_gpu_pvc"),
+    sycl::unittest::generateImageWithCompileTarget("SingleTaskKernel",
+                                                   "intel_gpu_skl"),
+    sycl::unittest::generateDefaultImage({"NDRangeKernel"}),
+    sycl::unittest::generateImageWithCompileTarget("NDRangeKernel",
+                                                   "spir64_x86_64"),
+    sycl::unittest::generateImageWithCompileTarget("NDRangeKernel",
+                                                   "intel_gpu_pvc"),
+    sycl::unittest::generateImageWithCompileTarget("NDRangeKernel",
+                                                   "intel_gpu_skl"),
+    sycl::unittest::generateDefaultImage({"RangeKernel"}),
+    sycl::unittest::generateImageWithCompileTarget("RangeKernel",
+                                                   "spir64_x86_64"),
+    sycl::unittest::generateImageWithCompileTarget("RangeKernel",
+                                                   "intel_gpu_pvc"),
+    sycl::unittest::generateImageWithCompileTarget("RangeKernel",
+                                                   "intel_gpu_skl"),
+    sycl::unittest::generateImageWithCompileTarget("NoDeviceKernel",
+                                                   "intel_gpu_bdw"),
+};
+
+static sycl::unittest::UrImageArray<std::size(Img)> ImgArray{Img};
```

Comment:
I can't find where this is used. Can you explain?

Reply:
That's most likely expected. If I'm not mistaken, `UrImageArray` constructor calls `__sycl_register_lib` and essentially makes those mock device images available to SYCL RT. Therefore, that variable is not directly used, but its data for sure used by the test.